### PR TITLE
[plugin.video.nhkworldtv@matrix] 1.0.2+matrix.1

### PR DESCRIPTION
**Pull request by sbroenne**
### Add-on details:

- **General**
  - Add-on name: NHK World TV
  - Add-on ID: plugin.video.nhkworldtv
  - Version number: 1.0.2+matrix.1
  - Kodi/repository version: matrix

- **Code location**
  - URL: https://github.com/sbroenne/plugin.video.nhkworldtv
  

Watch NHK World Japan in Full HD

- Live Stream 
- On demand programs - By program, latest and most watched, categories and playlist. You can also browse and search through all programs.
- Live schedule
- Top Stories and At a Glance
- News programs like NEWSLINE, NEWSLINE IN DEPTH and NEWSROOM TOKYO

Content is retrieved and played directly from the NHK World web site.

Plugin supports fallback to HD if you should encounter buffering issues.

Tested on Kodi 18.6+ Leia and 19 Matrix.
        

### Description of changes:

- Removed optional setting of View States so that plugin can be submitted to official repo
        

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.nhkworldtv/addon.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Watch NHK World Japan in Full HD[CR]
+NHK provides streams via Akamai in 720p and 1080p - the plug-in defaults to 1080p where available.[CR]
+If you encounter buffering issues, you can fall-back to 720p by changing the 'Use 720p instead of 1080p' addon-setting.[CR]
+Tested on Kodi 18.6+ and 19.[CR]
+        </description>
+        <disclaimer lang="en_GB">This plugin is a fan project and not related in any way to NHK!</disclaimer>
+        <platform>all</platform>
+        <license>GPL-3.0</license>
+        <forum></forum>
+        <language>en_GB</language>
```
Comment (enen92) [inline plugin.video.nhkworldtv/addon.xml]: Use only en not en_gb here

File: plugin.video.nhkworldtv/addon.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        <description lang="en_GB">
+Watch NHK World Japan in Full HD[CR]
+NHK provides streams via Akamai in 720p and 1080p - the plug-in defaults to 1080p where available.[CR]
+If you encounter buffering issues, you can fall-back to 720p by changing the 'Use 720p instead of 1080p' addon-setting.[CR]
+Tested on Kodi 18.6+ and 19.[CR]
+        </description>
+        <disclaimer lang="en_GB">This plugin is a fan project and not related in any way to NHK!</disclaimer>
+        <platform>all</platform>
+        <license>GPL-3.0</license>
+        <forum></forum>
```
Comment (enen92) [inline plugin.video.nhkworldtv/addon.xml]: Drop if empty

File: plugin.video.nhkworldtv/addon.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <summary lang="en_GB">Watch NHK World Japan in Full HD - live stream, on-demand and news programs</summary>
+        <description lang="en_GB">
+Watch NHK World Japan in Full HD[CR]
+NHK provides streams via Akamai in 720p and 1080p - the plug-in defaults to 1080p where available.[CR]
+If you encounter buffering issues, you can fall-back to 720p by changing the 'Use 720p instead of 1080p' addon-setting.[CR]
+Tested on Kodi 18.6+ and 19.[CR]
+        </description>
+        <disclaimer lang="en_GB">This plugin is a fan project and not related in any way to NHK!</disclaimer>
+        <platform>all</platform>
+        <license>GPL-3.0</license>
```
Comment (enen92) [inline plugin.video.nhkworldtv/addon.xml]: This should be "GPL-3.0-only" or "GPL-3.0-or-later" depending on how you license your plugin (valid spdx identifiers)

File: plugin.video.nhkworldtv/lib/episode.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    @cast.setter
+    def cast(self, value):
+        """ Sets the cast info label """
+        self._cast = value
+
+    @property
+    def kodi_list_item(self):
+        """ Gets the current Kodi List Item """
+
+        if (self.url is not None):
```
Comment (enen92) [inline plugin.video.nhkworldtv/lib/episode.py:190]: You might want to define offscreen=True to speed things up. See https://codedocs.xyz/xbmc/xbmc/group__python__xbmcgui__listitem.html#ga15bcb930a4f91dd97f8ed7c633a20618 . This change is just a recommendation.

File: plugin.video.nhkworldtv/lib/first_run_wizard.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    if (not ADDON.getSettingBool('run_wizard')):
+        return (False)
+
+    xbmc.log("Starting Set-up Wizard")
+    dialog = xbmcgui.Dialog()
+
+    delimiter = '\n\n'
+    lines = []
+    lines.append('NHK World TV comes with a set of configurable options.')
+    lines.append('You can always change them in the addon-settings later.')
```
Comment (enen92) [inline plugin.video.nhkworldtv/lib/first_run_wizard.py]: Sorry, strings shown in the UI have to be localized, please move them to strings.po

File: plugin.video.nhkworldtv/lib/first_run_wizard.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    abort_wizard = dialog.yesno('1 of 3: Set-up Wizard - Welcome',
+                                delimiter.join(lines),
+                                nolabel='Yes, continue set-up',
+                                yeslabel='No, I am good')
+
+    if (abort_wizard):
+        ADDON.setSettingBool('run_wizard', False)
+        return (False)
+
+    lines = []
```
Comment (enen92) [inline plugin.video.nhkworldtv/lib/first_run_wizard.py]: Same applies to all below

File: plugin.video.nhkworldtv/lib/kodiutils.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    # Setting view mode - needs to be enabled in settings
+    if (ADDON.getSettingBool('set_view_mode')):
+        # Change view mode
+        xbmc.log('Switching to View Mode: {0}'.format(
+            VIEW_MODES_REVERSE[view_mode]))
+        # Set the content
+        if (view_mode != VIEW_MODE_WIDELIST):
+            # Do not set the content typ for WideList
+            xbmcplugin.setContent(plugin_handle, content_type)
+        xbmc.executebuiltin('Container.SetViewMode({0})'.format(view_mode))
```
Comment (enen92) [inline plugin.video.nhkworldtv/lib/kodiutils.py]: This is not allowed in the repo, sorry

**Review by enen92 — CHANGES_REQUESTED** [review]
See comms

File: plugin.video.nhkworldtv/lib/kodiutils.py (hunk 7 above)
Comment (sbroenne) [reply plugin.video.nhkworldtv/lib/kodiutils.py]: This is **not enabled by default** in the plug-in.  You need to **enable** this in the add-on settings. Another way is to actively select a non-default answer in the First-run wizard.

`<setting id="set_view_mode" type="bool" label="30005" default="false"/>`

I can remove it completely but would prefer not to (I use this to  *mirror* the UI on the NHK web site.)

File: plugin.video.nhkworldtv/lib/episode.py (hunk 4 above)
Comment (sbroenne) [reply plugin.video.nhkworldtv/lib/episode.py:190]: Thank you for the advice - I have changed the code.

File: plugin.video.nhkworldtv/lib/kodiutils.py (hunk 7 above)
Comment (sbroenne) [reply plugin.video.nhkworldtv/lib/kodiutils.py]: @enen92: I have now removed all settings of ViewMode and have adapted some other parts of the plugin. Could you please review again?

**Comment by enen92** [comment]
Thanks for adjusting @sbroenne and for your work on the addon. Unfortunately it was brought to my attention that we already provide another nhkworld addon in matrix (actually 2 in leia) but our policy has always been to keep only one active addon per service in the repo to avoid confusion for users.
For this reason, and since I don't want to be involved in such politics nor do I know this service at all, I'll remove myself from the review of this addon and request the review and further progress from other team members.
Sorry if somehow this has made you waste your time.

**Review by enen92 — COMMENTED** [review]
@learningit @Lunatixz @ronie can you guys do the follow up here?
I am out of this review.

**Comment by sbroenne** [comment]
Thank you @enen92  for the review.. It made the plugin better so it was not a waste of my time and I hope it wasn't waste of yours.   Thank you!

**Comment by ronie** [comment]
thanx for the submission, but sadly we'll have to close this for the reason given by @enen92.

**Pull request closed without merging**